Sprint-1/implement/sum.test.js

```diff
@@ -1,36 +1,32 @@
-/* Sum the numbers in an array
+const sum = require("./sum.js");
 
-In this kata, you will need to implement a function that sums the numerical elements of an array
 
-E.g. sum([10, 20, 30]), target output: 60
-E.g. sum(['hey', 10, 'hi', 60, 10]), target output: 80 (ignore any non-numerical elements)
-*/
+test("given an empty array, returns 0", () => {
+  expect(sum([])).toBe(0);
+});
 
-const sum = require("./sum.js");
 
-// Acceptance Criteria:
+test("given an array with just one number, returns that number", () => {
+  expect(sum([42])).toBe(42);
+});
+
+
+test("given an array containing negative numbers, returns the correct total sum", () => {
+  expect(sum([-10, 20, -30, 40])).toBe(20);
+});
+
 
-// Given an empty array
-// When passed to the sum function
-// Then it should return 0
-test.todo("given an empty array, returns 0")
+test("given an array with decimal/float numbers, returns the correct total sum", () => {
+  expect(sum([1.5, 2.5, 3.5])).toBe(7.5);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Reply:
For floating-point numbers, use an epsilon-based comparison to account for the inaccuracies inherent in floating-point arithmetic. This ensures more reliable tests for equality.